samples/boards/stm32/i2c_config/src/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+
+
+#if DT_NODE_HAS_STATUS(DT_ALIAS(i2c_0), okay)
+#define I2C_DEV_NODE	DT_ALIAS(i2c_0)
+#elif DT_NODE_HAS_STATUS(DT_ALIAS(i2c_1), okay)
+#define I2C_DEV_NODE	DT_ALIAS(i2c_1)
+#elif DT_NODE_HAS_STATUS(DT_ALIAS(i2c_2), okay)
+#define I2C_DEV_NODE	DT_ALIAS(i2c_2)
+#else
+#error "Please set the correct I2C device"
+#endif
+
+int main(void)
+{
+	const struct device *const i2c_dev = DEVICE_DT_GET(I2C_DEV_NODE);
+	uint32_t bit_rate, cfg_timings;
+
+	if (!device_is_ready(i2c_dev)) {
+		printk("I2C device is not ready\n");
+		return -1;
+	}
+
+	/* 2. Verify i2c_get_config() */
+	if (i2c_get_config(i2c_dev, (uint32_t *)&cfg_timings)) {
```

Comment:
i2c_get_config will return `data->dev_config`, not `cfg_timings`

Reply:
yes, the uint32_t config